statuspage/src/store/index.js

```diff
@@ -135,7 +135,11 @@ export default new Vuex.Store({
                     group[check.namespace][groupBy] = {
                         type: check.type,
                         namespace: check.namespace,
```

Comment:
For syntax see:
- Nullish coalescing operator (??): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
- Object Destructuring: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

We could approach this section in a couple of ways:

In the first, we're setting defaults with the nullish coalescing operator (if we want defaults).

```
   const { type, namespace, name, interval } = check;

      group[namespace][groupBy] = {
          type: type ?? "",
          namespace: namespace ?? "",
          name: name ?? "",
          interval: interval ?? 0,
          items: []
      }
```

But if we're confident of the values, we might do the following:

```
   const { type, namespace, name, interval } = check;
   group[namespace][groupBy] = { type, namespace, name, interval, items: [] }    
```

If we destructure at the top of the loop, we can use the variables all the way down, also, so:

`        
if (group[check.namespace] == null) {
            group[check.namespace] = {}
 }
`
                
Could be, for example:

`        
 group[namespace] = group[namespace] ?? {}
`

And:


```
if (group[check.namespace][groupBy] == null) {
      group[check.namespace][groupBy] = {
          type: check.type,
          namespace: check.namespace,
          name: check.name,
          interval: check.interval,
          severity: check.severity,
          owner:    check.owner,
          description: check.description,
          items: []
      }
}
```


Could be, at its most terse:

`
group[namespace][groupBy] = group[namespace][groupBy] ?? { type, namespace, name, interval, items: [] }
`

Reply:
For now changed to use `Nullish coalescing operator` maybe the possibility to use `reduce` can be explored when we will refactor this whole component